docs/DeveloperGuide.md

```diff
@@ -1,34 +1,179 @@
+
+### Audience 
+The design documentation is in general for anyone who wants to understand the system architecture and design of 
+PAC. The following groups are in particular the intended audience of the document.
+- PAC project managers
+- PAC developers
+- PAC software testers
+
+## Architecture
+This section presents the architecture of PAC. It explains the architecture of main components of PAC.
+
+### Overall Architecture
+*Overall Class diagram*
+{To be added in future revisions}
+
+### Command
+![Command](images/Command.png "Class diagram of Command component")         
+*Class diagram of the Command component*  
+
+Commands are the main classes to be executed in PAC. All of the specific Command classes inherit the 
+base Command abstract class, and utilize its abstract execute() method.  
+A subclass of Command is created and executed when the professor input a corresponding command.
+ 
+### Parser
+*Class diagram of the Parser component*  
+There are total of four Parser classes as shown below. Each Parser class correspond to a feature 
+of PAC. 
+
+| Parser                    | Created in                                                    |
+|---------------------------|---------------------------------------------------------------|
+| EventParser               | EventCommandInterpreter                                       | 
+| CalenderParser            | EventCommandInterpreter                                       | 
+| AttendanceParser          | Step-by-step command at Attendance-related command classes    |
+| PerformanceParser         | Step-by-step command at performance-related command classes   |  
+
+A Parser class is created when a user input contains data to be stored or used in certain features.    
 
-## Design & Implementation
+### UI
+![Ui](images/Ui.png "Class diagram of Ui component")                
+*Class diagram of the UI component*   
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+UI is the main class handles user display, which includes reading user input and printing information 
+back to the user on command-line.  
+Besides the normal command line messages, there are two subclasses of UI: 
+DisplayList and DisplayTable, to specifically print the list and table interface to professor. 
 
+## Feature Design and Implementation 
+### Event
+![event](images/event.png "Class diagram of Event component")           
+*Class diagram of the Event component*
+
+1. When a user enters an event-related command, the command is analysed by `EventCommandInterpreter`. 
+1. Once determined, the relevant information (e.g. index, name, time, date, venue) are extracted by 
+`EventParser`.
+1. Then, the relevant class that corresponds to the command is created, with the information extracted 
+from the previous step passed into it. It modifies `Event` or `EventList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
```

Comment:
The diagram looks neat and very informative. Perhaps more description would aid the user a better understanding of the diagram.